BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -8,10 +9,183 @@ public class BullsAndCowsGameTest
+        public void Should_return_result_when_all_digits_correct_but_some_positions_wrong(string guess, string secret,
+            string bullcows)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secret);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+
+            //when
+            string answer = game.Guess(guess);
+
+            //then
+            Assert.Equal(bullcows, answer);
+        }
+
+        [Theory]
+        [InlineData("2 1 5 6", "1234", "0A2B")]
+        [InlineData("6 1 5 8", "1234", "0A1B")]
+        [InlineData("3 1 2 6", "1234", "0A3B")]
+        public void Should_return_result_when_some_digits_correct_and_all_positions_wrong(string guess, string secret,
+            string bullcows)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secret);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+
+            //when
+            string answer = game.Guess(guess);
+
+            //then
+            Assert.Equal(bullcows, answer);
+        }
+
+        [Theory]
+        [InlineData("1 5 7 8", "1234", "1A0B")]
+        [InlineData("1 2 5 6", "1234", "2A0B")]
+        [InlineData("1 2 5 3", "1234", "2A1B")]
+        [InlineData("1 4 2 5", "1234", "1A2B")]
+        public void Should_return_result_when_some_digits_correct_and_some_positions_correct(string guess, string secret,
+            string bullcows)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secret);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+
+            //when
+            string answer = game.Guess(guess);
+
+            //then
+            Assert.Equal(bullcows, answer);
+        }
+
+        [Theory]
+        [InlineData("1 2 3 4 5")]
+        [InlineData("1 2 3")]
+        public void Should_return_wrong_input_tip_when_number_of_digits_is_not_4(string guess)
+        {
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+
+            string answer = game.Guess(guess);
+
+            Assert.Equal("Wrong Input, input again", answer);
+        }
+
+        [Theory]
+        [InlineData("1 2 3 1")]
+        [InlineData("1 1 1 1")]
+        public void Should_return_wrong_input_tip_when_some_digits_are_same(string guess)
+        {
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+
+            string answer = game.Guess(guess);
+
+            Assert.Equal("Wrong Input, input again", answer);
+        }
+
+        [Theory]
+        [InlineData("1 2 23 4")]
```

Comment:
should add cases like "123" "1 s" whose length is not long enough

Reply:
`Should_return_wrong_input_tip_when_number_of_digits_is_not_4`  this test method checked this kind of case

Reply:
> should add cases like "123" "1 s" whose length is not long enough

`Should_return_wrong_input_tip_when_number_of_digits_is_not_4` this test method checked this kind of case